_posts/2020-05-02-schedule-your-functions.md

```diff
@@ -0,0 +1,264 @@
+---
+title: Keep your functions running to schedule with the cron-connector.
+description: "In this post Martin will show you how to schedule your functions with the cron-connector, for regular tasks and scheduled jobs"
+date: 2020-05-02
+image: /images/2020-05-02-schedule-your-functions/accuracy-alarm-clock-analogue.jpg
+categories:
+  - cron
+  - kubernetes
+  - serverless
+  - linux
+  - tutorial
+  - example
+author_staff_member: martin
+dark_background: true
+
+---
+
+In this post Martin will show you how schedule your functions with the cron-connector, for regular tasks and scheduled jobs.
+
+One of the things we need very often is our functions to be executed on a schedule. This is where the cron-connector shines. We will showcase how you can use it in this post, making sure your functions always run on time.
+
+## Motivation
+
+OpenFaaS provides a very convenient and flexible way to quickly write and expose your code to be used by external systems. Often times those systems don't have a hook to which you can schedule function execution. Also often times we need to run those functions without external systems, we just want our code to be ran on an interval.
+
+For those purposes [connector-sdk](https://github.com/openfaas-incubator/connector-sdk) was created. It provides the means to quickly adapt different systems to invoke functions. Some of them are [Kafka](https://github.com/openfaas-incubator/kafka-connector), [NATS](https://github.com/openfaas-incubator/nats-connector), [AWS SNS](https://github.com/affix/OpenFaaS-SNS) and [many more](https://docs.openfaas.com/reference/triggers/).
+
+Now we need our functions to to be invoked on an interval. In Linux we have the [cron](https://en.wikipedia.org/wiki/Cron) that allows us to execute code on a schedule. In Kubernetes, this was abstracted into the `CronJob` object.  For any OpenFaaS cluster, the [cron-connector](https://github.com/openfaas-incubator/cron-connector) uses the `connector-sdk` to provide the same functionality, running our functions on a predefined schedule.
```

Comment:
In Linux we have the [cron](https://en.wikipedia.org/wiki/Cron) that allows us to execute code on a schedule

"in Linux we have" doesn't sound quite right:

[cron](https://en.wikipedia.org/wiki/Cron) is often used on GNU/Linux systems to execute tasks to a pre-defined schedule.

Reply:
You are right will rephrase :+1:

Reply:
Rephrased to `Cron is often used on GNU/Linux systems to execute tasks to a pre-defined schedule.` :heavy_check_mark: